test cases/common/139 override options/meson.build

```diff
@@ -0,0 +1,6 @@
+project('option override', 'c',
+  default_options : 'unity=true')
+
+executable('mustunity', 'one.c', 'two.c')
+executable('notunity', 'three.c', 'four.c',
+  override_options : ['unity=false'])
```

Comment:
Should test the other option types too, not just string. And maybe also test overriding options set from various sources: project(), built-in, command-line, meson_options, etc.